app/code/community/Facebook/AdsExtension/Model/FBProductFeed.php

```diff
@@ -746,7 +746,12 @@ private function getFinalPrice($product, $price = null) {
     if ($price === null) {
       $price = $product->getData('special_price');
     }
-    return $this->taxHelper->getPrice($product, $price);
+    $store = Mage::app()->getStore(1);
+    $taxCalculation = Mage::getModel('tax/calculation');
+    $request = $taxCalculation->getRateRequest(null, null, null, $store);
+    $taxClassId = $product->getTaxClassId();
+    $percent = $taxCalculation->getRate($request->setProductClassId($taxClassId));
+    return $this->taxHelper->getPrice($product, $price) + (($this->taxHelper->getPrice($product, $price) * $percent) / 100);
```

Comment:
I think we should only do this if the store setting is display with and without VAT, if possible.

If we do this in other cases it may introduce problems as mentioned in my main comment.
(1) If the store has set to no VAT, we will apply VAT and VAT will be shown on FB, which is not good.
(2) If the store has set to VAT, then taxHelper will already apply the VAT, so this code may apply  the VAT again after it has already been applied, which is not good.